src/proc/interface.rs

```diff
@@ -145,16 +150,44 @@ impl<'a> Interface<'a> {
                             crate::Expression::AccessIndex { base, .. } => {
```

Comment:
also, I see a bug here: all the expressions starting from `pointer` down to the last element in this sub-tree should be added to `self.add_inputs`